strutil/strutil.go

```diff
@@ -511,3 +512,37 @@ func Reverse(in string) string {
 	}
 	return string(out)
 }
+
+// ReplaceNonMatcher replaces all characters within a given that do not match
+// a given regular expression. The function returns the updated string and the
+// number of replacements
+func ReplaceNonMatcher(in, matcher string, replaceWith byte) (string, int) {
+	var adjustments int
+	work := []byte(in)
+	allowList := regexp.MustCompile(matcher)
```

Comment:
Please adjust the function signature to return an error and use Compile. Using MustCompile within this function is dangerous as it can trigger a panic; MustCompile is safest when run in an `init()` function so you panic at startup rather than at some point during runtime. Since the regex isn't known in advance it can't be moved to an `init()` function and this should instead return an error.

Reply:
Great call, will do so!